ravenframework/CodeInterfaceClasses/MELCOR/melcorCombinedInterface.py

```diff
@@ -13,104 +13,189 @@
     """
     foundMelcorInp = False
-    for index, inputFile in enumerate(inputFiles):
+    for index, inputFile in enumerate(currentInputFiles):
       if inputFile.getExt() in self.getInputExtension():
         foundMelcorInp = True
-        melgIn = inputFiles[index]
-        melcIn = inputFiles[index]
+        melgIn = currentInputFiles[index]
+        melcIn = currentInputFiles[index]
     if not foundMelcorInp:
       raise IOError("Unknown input extensions. Expected input file extensions are "+ ",".join(self.getInputExtension())+" No input file has been found!")
-    return melgIn,melcIn
+    return melgIn, melcIn
+
 
   def generateCommand(self, inputFiles, executable, clargs=None, fargs=None, preExec=None):
     """
-      Generate a command to run MELCOR (combined MELGEN AND MELCOR)
-      Collects all the clargs and the executable to produce the command-line call.
+      This method is used to retrieve the command (in tuple format) needed to launch the Code.
+      See base class.  Collects all the clargs and the executable to produce the command-line call.
       Returns tuple of commands and base file name for run.
       Commands are a list of tuples, indicating parallel/serial and the execution command to use.
-      This method is used to retrieve the command (in tuple format) needed to launch the Code.
-      @ In, inputFiles, list, List of input files (length of the list depends on the number of inputs that have been added in the Step is running this code)
+      @ In, inputFiles, list, List of input files (length of the list depends on the number of inputs have been added in the Step is running this code)
       @ In, executable, string, executable name with absolute path (e.g. /home/path_to_executable/code.exe)
       @ In, clargs, dict, optional, dictionary containing the command-line flags the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 i0extension =0 .inp0/ >< /Code >)
-      @ In, fargs, dict, optional, a dictionary containing the axuiliary input file variables the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 aux0extension =0 .aux0/ >< /Code >)
+      @ In, fargs, dict, optional, a dictionary containing the auxiliary input file variables the user can specify in the input (e.g. under the node < Code >< fileargstype =0 input0arg =0 aux0extension =0 .aux0/ >< /Code >)
       @ In, preExec, string, optional, a string the command that needs to be pre-executed before the actual command here defined
       @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the code (string), returnCommand[1] is the name of the output root
     """
-    if preExec is None:
-      raise IOError('No preExec listed in input! Exiting...')
+    found = False
+
+    for index, inputFile in enumerate(inputFiles):
+      if inputFile.getExt() in self.getInputExtension():
```

Comment:
I do not see a function to add the input extension for this interface. I think you need to update the self.inputExtensions using setInputExtension method or using addDefaultExtension method.

Reply:
It seems this issue has not been addressed yet.